File: Leetcode - Java/905_array_sortArrayByParity.java
// https://leetcode.com/problems/sort-array-by-parity/

class Solution {
    public int[] sortArrayByParity(int[] A) {
        int left = 0, right = A.length-1; 
        
        while(left < right) {
            if(A[left]%2 > A[right]%2) {
                int temp = A[left];
                A[left] = A[right];
                A[right] = temp; 
            }
            
            if(A[left]%2 == 0) left++;
            if(A[right]%2 == 1) right--;
        }
        return A;
    }
}

File: Leetcode - Python/905_array_sortArrayByParity.py
# https://leetcode.com/problems/sort-array-by-parity/


class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        left, right = 0, len(A) - 1
        while left < right:
            if A[left] % 2 > A[right] % 2:
                A[left], A[right] = A[right], A[left]
            if A[left] % 2 == 0:
                left += 1
            if A[right] % 2 == 1:
                right -= 1

        return A
